src/lib/containers/ForumSearch.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useEffect, useState } from 'react'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faSearch, faTimes } from '@fortawesome/free-solid-svg-icons'
+import { forumSearch, getEntity } from '../utils/SynapseClient'
+import { DiscussionSearchResponse, Match } from '../utils/synapseTypes/DiscussionSearch'
+import { useSynapseContext } from '../utils/SynapseContext'
+import DiscussionReply from './DiscussionSearchResult'
+import { SynapseSpinner } from './LoadingScreen'
+import { Entity } from '../utils/synapseTypes'
+import Typography from '../utils/typography/Typography'
+import NoSearchResults from '../assets/icons/NoSearchResults'
+import { Button } from 'react-bootstrap'
+
+export type ForumSearchProps = {
+    forumId: string,
+    projectId?: string,
```

Comment:
More comments throughout.
If projectId is set, will show that these search results are associated with the given project.